### PR TITLE
feat: adding schema comments and column cleanup

### PR DESCRIPTION
**Pull request by amenowanna**
The purpose of this change is to add comments to our schema for documentation. As part of this process some column cleanup was done as well.

### CODE REVIEW

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.trivy_repo_vulnerabilities.vulnerability_installed_version IS 'vulnerability installed version';
+COMMENT ON COLUMN public.trivy_repo_vulnerabilities.vulnerability_severity IS 'vulnerability severity';
+COMMENT ON COLUMN public.trivy_repo_vulnerabilities.vulnerability_title IS 'vulnerability title';
+COMMENT ON COLUMN public.trivy_repo_vulnerabilities.vulnerability_description IS 'vulnerability description';
+COMMENT ON COLUMN public.trivy_repo_vulnerabilities._mergestat_synced_at IS 'timestamp when record was synced into the MergeStat database';
+
+COMMENT ON TABLE public.schema_migrations IS 'MergeStat internal table to track schema migrations';
+COMMENT ON TABLE public.schema_migrations_history IS 'MergeStat internal table to track schema migrations history';
+
+COMMIT;
```
Comment (ramiroJCB) [inline migrations/202210131714933_add_schema_comments.up.sql:281]: This are a lot of comments Oo

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.git_blame._mergestat_synced_at IS 'timestamp when record was synced into the MergeStat database';
+
+COMMENT ON TABLE public.git_commit_stats IS 'git commit stats of a repo';
+COMMENT ON COLUMN public.git_commit_stats.repo_id IS 'foreign key for public.repos.id';
+COMMENT ON COLUMN public.git_commit_stats.commit_hash IS 'hash of the commit';
+COMMENT ON COLUMN public.git_commit_stats.file_path IS 'path of the file the modification was made in';
+COMMENT ON COLUMN public.git_commit_stats.additions IS 'the number of additions in this path of the commit';
+COMMENT ON COLUMN public.git_commit_stats.deletions IS 'the number of deletions in this path of the commit';
+COMMENT ON COLUMN public.git_commit_stats.old_file_mode IS 'old file mode derived from git mode. possible values (unknown,none,regular_file,symbolic_link,git_link)';
+COMMENT ON COLUMN public.git_commit_stats.new_file_mode IS 'new file mode derived from git mode. possible values (unknown,none,regular_file,symbolic_link,git_link)';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:19]:
```suggestion
COMMENT ON COLUMN public.git_commit_stats.old_file_mode IS 'old file mode derived from git mode. possible values (unknown, none, regular_file, symbolic_link, git_link)';
COMMENT ON COLUMN public.git_commit_stats.new_file_mode IS 'new file mode derived from git mode. possible values (unknown, none, regular_file, symbolic_link, git_link)';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.git_files._mergestat_synced_at IS 'timestamp when record was synced into the MergeStat database';
+
+COMMENT ON TABLE public.git_refs IS 'git refs of a repo';
+COMMENT ON COLUMN public.git_refs.repo_id IS 'foreign key for public.repos.id';
+COMMENT ON COLUMN public.git_refs.hash IS 'hash of the commit for refs that are not of type tag';
+COMMENT ON COLUMN public.git_refs.name IS 'name of the ref';
+COMMENT ON COLUMN public.git_refs.remote IS 'remote of the ref';
+COMMENT ON COLUMN public.git_refs.target IS 'target of the ref';
+COMMENT ON COLUMN public.git_refs.type IS 'type of the ref';
+COMMENT ON COLUMN public.git_refs.tag_commit_hash IS 'hash of the commit for refs that or of type tag';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:49]:
```suggestion
COMMENT ON COLUMN public.git_refs.tag_commit_hash IS 'hash of the commit for refs that are of type tag';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.git_refs._mergestat_synced_at IS 'timestamp when record was synced into the MergeStat database';
+
+COMMENT ON view public.git_branches IS 'view of git refs of type branch of a repo';
+COMMENT ON COLUMN public.git_branches.repo_id IS 'foreign key for public.repos.id';
+COMMENT ON COLUMN public.git_branches.hash IS 'hash of the commit for refs that are not of type tag';
+COMMENT ON COLUMN public.git_branches.name IS 'name of the ref';
+COMMENT ON COLUMN public.git_branches.remote IS 'remote of the ref';
+COMMENT ON COLUMN public.git_branches.target IS 'target of the ref';
+COMMENT ON COLUMN public.git_branches.type IS 'type of the ref';
+COMMENT ON COLUMN public.git_branches.tag_commit_hash IS 'hash of the commit for refs that or of type tag';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:59]:
```suggestion
COMMENT ON COLUMN public.git_branches.tag_commit_hash IS 'hash of the commit for refs that are of type tag';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.git_branches._mergestat_synced_at IS 'timestamp when record was synced into the MergeStat database';
+
+COMMENT ON view public.git_tags IS 'view of git refs of type tag of a repo';
+COMMENT ON COLUMN public.git_tags.repo_id IS 'foreign key for public.repos.id';
+COMMENT ON COLUMN public.git_tags.hash IS 'hash of the commit for refs that are not of type tag';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:64]:
```suggestion
COMMENT ON COLUMN public.git_tags.hash IS 'hash of the ref that are of type tag';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.git_tags.name IS 'name of the ref';
+COMMENT ON COLUMN public.git_tags.remote IS 'remote of the ref';
+COMMENT ON COLUMN public.git_tags.target IS 'target of the ref';
+COMMENT ON COLUMN public.git_tags.type IS 'type of the ref';
+COMMENT ON COLUMN public.git_tags.tag_commit_hash IS 'hash of the commit for refs that or of type tag';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:69]:
```suggestion
COMMENT ON COLUMN public.git_tags.tag_commit_hash IS 'hash of the commit for refs that are of type tag';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON TABLE public.github_issues IS 'issues of a GitHub repo';
+COMMENT ON COLUMN public.github_issues.repo_id IS 'foreign key for public.repos.id';
+COMMENT ON COLUMN public.github_issues.author_login IS 'login of the author of the issue';
+COMMENT ON COLUMN public.github_issues.body IS 'body of the issue';
+COMMENT ON COLUMN public.github_issues.closed IS 'boolean to determine if the issue is closed';
+COMMENT ON COLUMN public.github_issues.closed_at IS 'timestamp of when the issue was closed';
+COMMENT ON COLUMN public.github_issues.created_via_email IS 'boolean to determine if the issue was created via email';
+COMMENT ON COLUMN public.github_issues.database_id IS 'GitHub database_id of the issue';
+COMMENT ON COLUMN public.github_issues.editor_login IS 'login of the editor of the issue';
+COMMENT ON COLUMN public.github_issues.includes_created_edit IS 'boolean to determine if the issue was edited and includes and edit with the creation data';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:81]:
```suggestion
COMMENT ON COLUMN public.github_issues.includes_created_edit IS 'boolean to determine if the issue was edited and includes and edit with the creation data';
```

not sure of a good phrasing exactly - unclear a bit on what the column actually is too 😄, I will look it up in the GitHub API too

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.github_pull_request_commits.hash IS 'hash of the commit';
+COMMENT ON COLUMN public.github_pull_request_commits.message IS 'message of the commit';
+COMMENT ON COLUMN public.github_pull_request_commits.author_email IS 'email of the author of the modification';
+COMMENT ON COLUMN public.github_pull_request_commits.author_name IS 'name of the author of the the modification';
+COMMENT ON COLUMN public.github_pull_request_commits.author_when IS 'timestamp of when the modifcation was authored';
+COMMENT ON COLUMN public.github_pull_request_commits.committer_email IS 'email of the author who committed the modification';
+COMMENT ON COLUMN public.github_pull_request_commits.committer_name IS 'name of the author who committed the modification';
+COMMENT ON COLUMN public.github_pull_request_commits.committer_when IS 'timestamp of when the commit was made';
+COMMENT ON COLUMN public.github_pull_request_commits.additions IS 'the number of additions in this path of the commit';
+COMMENT ON COLUMN public.github_pull_request_commits.deletions IS 'the number of deletions in this path of the commit';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:109]:
```suggestion
COMMENT ON COLUMN public.github_pull_request_commits.additions IS 'the number of additions in the commit';
COMMENT ON COLUMN public.github_pull_request_commits.deletions IS 'the number of deletions in the commit';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+
+COMMENT ON TABLE public.github_pull_requests IS 'pull requests of a GitHub repo';
+COMMENT ON COLUMN public.github_pull_requests.repo_id IS 'foreign key for public.repos.id';
+COMMENT ON COLUMN public.github_pull_requests.additions IS 'the number of additions in the pull request';
+COMMENT ON COLUMN public.github_pull_requests.deletions IS 'the number of deletions in the pull request';
+COMMENT ON COLUMN public.github_pull_requests.author_login IS 'login of the author of the pull request';
+COMMENT ON COLUMN public.github_pull_requests.author_association IS 'author association to the repo';
+COMMENT ON COLUMN public.github_pull_requests.author_avatar_url IS 'URL to the avatar of the author of the pull request';
+COMMENT ON COLUMN public.github_pull_requests.author_name IS 'name of the author of the pull request';
+COMMENT ON COLUMN public.github_pull_requests.base_ref_oid IS 'the base ref associated with the pull request';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:142]:
```suggestion
COMMENT ON COLUMN public.github_pull_requests.base_ref_oid IS 'the base ref object id associated with the pull request';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.github_pull_requests.body IS 'body of the pull request';
+COMMENT ON COLUMN public.github_pull_requests.changed_files IS 'the number of files changed/modified in the pull request';
+COMMENT ON COLUMN public.github_pull_requests.closed IS 'boolean to determine if the pull request is closed';
+COMMENT ON COLUMN public.github_pull_requests.closed_at IS 'timestamp of when the pull request was closed';
+COMMENT ON COLUMN public.github_pull_requests.comment_count IS 'the number of comments in the pull request';
+COMMENT ON COLUMN public.github_pull_requests.created_at IS 'timestamp of when the pull request was created';
+COMMENT ON COLUMN public.github_pull_requests.created_via_email IS 'boolean to determine if the pull request was created via email';
+COMMENT ON COLUMN public.github_pull_requests.database_id IS 'GitHub database_id of the pull request';
+COMMENT ON COLUMN public.github_pull_requests.editor_login IS 'login of the editor of the pull request';
+COMMENT ON COLUMN public.github_pull_requests.head_ref_oid IS 'the head ref associated with the pull request';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:154]:
```suggestion
COMMENT ON COLUMN public.github_pull_requests.head_ref_oid IS 'the head ref object id associated with the pull request';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.github_pull_requests.head_ref_name IS 'the name of head ref associated with the pull request';
+COMMENT ON COLUMN public.github_pull_requests.head_repository_name IS 'the name of the head repo for the pull request';
+COMMENT ON COLUMN public.github_pull_requests.is_draft IS 'boolean to determine if the pull request is a draft';
+COMMENT ON COLUMN public.github_pull_requests.label_count IS 'number of labels associated to the pull request';
+COMMENT ON COLUMN public.github_pull_requests.last_edited_at IS 'timestamp of when the pull request was last edited';
+COMMENT ON COLUMN public.github_pull_requests.locked IS 'boolean to determine if the pull request is locked';
+COMMENT ON COLUMN public.github_pull_requests.maintainer_can_modify IS 'boolean to determine if a maintaner can modify the pull request';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:161]:
```suggestion
COMMENT ON COLUMN public.github_pull_requests.maintainer_can_modify IS 'boolean to determine if a maintainer can modify the pull request';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+ALTER TABLE github_pull_requests
+DROP COLUMN IF EXISTS mantainer_can_modify;
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:163]: do we need to re-add this column in a subsequent migration?

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.github_repo_info.is_disabled IS 'boolean to determine if the repo is disabled';
+COMMENT ON COLUMN public.github_repo_info.is_mirror IS 'boolean to determine if the repo is a mirror';
+COMMENT ON COLUMN public.github_repo_info.is_private IS 'boolean to determine if the repo is private';
+COMMENT ON COLUMN public.github_repo_info.total_issues_count IS 'number of issues associated to the repo';
+COMMENT ON COLUMN public.github_repo_info.latest_release_author IS 'the author of the latest release in the repo';
+COMMENT ON COLUMN public.github_repo_info.latest_release_created_at IS 'timestamp of the creation of the latest release in the repo';
+COMMENT ON COLUMN public.github_repo_info.latest_release_name IS 'the name of the latest release in the repo';
+COMMENT ON COLUMN public.github_repo_info.latest_release_published_at IS 'timestamp of the publishing of the latest release in the repo';
+COMMENT ON COLUMN public.github_repo_info.license_key IS 'the licence key for the repo';
+COMMENT ON COLUMN public.github_repo_info.license_name IS 'the licence name for the repo';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:199]:
```suggestion
COMMENT ON COLUMN public.github_repo_info.license_key IS 'the license key for the repo';
COMMENT ON COLUMN public.github_repo_info.license_name IS 'the license name for the repo';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.github_repo_info.license_nickname IS 'the licence nickname for the repo';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:200]:
```suggestion
COMMENT ON COLUMN public.github_repo_info.license_nickname IS 'the license nickname for the repo';
```

File: migrations/202210131714933_add_schema_comments.up.sql [hunk 15]
```diff
@@ -0,0 +1,281 @@
...
+COMMENT ON COLUMN public.github_stargazers._mergestat_synced_at IS 'timestamp when record was synced into the MergeStat database';
+
+COMMENT ON TABLE public.repos IS 'git repositories to track';
+COMMENT ON COLUMN public.repos.id IS 'MergeStat identifier for the repo';
+COMMENT ON COLUMN public.repos.repo IS 'URL for the repo';
+COMMENT ON COLUMN public.repos.ref IS 'ref for the repo';
+COMMENT ON COLUMN public.repos.is_github IS 'boolean to determine if the repo is in GitHub';
+COMMENT ON COLUMN public.repos.created_at IS 'timestamp of when the MergeStat repo entry was created';
+COMMENT ON COLUMN public.repos.settings IS 'JSON settings for the repo';
+COMMENT ON COLUMN public.repos.tags IS 'array of tags for the repo for tags in GitHub as well as tags added in MergeStat';
```
Comment (patrickdevivo) [inline migrations/202210131714933_add_schema_comments.up.sql:233]:
```suggestion
COMMENT ON COLUMN public.repos.tags IS 'array of tags for the repo for topics in GitHub as well as tags added in MergeStat';
```

**Review by patrickdevivo — CHANGES_REQUESTED** [review]

File: migrations/202210131714933_add_schema_comments.up.sql (hunk 7 above)
Comment (amenowanna) [reply migrations/202210131714933_add_schema_comments.up.sql:81]: This is the wording in the [GitHub docs](https://docs.github.com/en/graphql/reference/objects)
`Check if this comment was edited and includes an edit with the creation data.`

File: migrations/202210131714933_add_schema_comments.up.sql (hunk 11 above)
Comment (amenowanna) [reply migrations/202210131714933_add_schema_comments.up.sql:161]: we don't know how to spell maintainer :)

File: migrations/202210131714933_add_schema_comments.up.sql (hunk 14 above)
Comment (amenowanna) [reply migrations/202210131714933_add_schema_comments.up.sql:200]: I have never been able to spell that correctly 🤦

File: migrations/202210131714933_add_schema_comments.up.sql (hunk 12 above)
Comment (amenowanna) [reply migrations/202210131714933_add_schema_comments.up.sql:163]: No the column exists with the correct spelling and is the comment right above this one.

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**